sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -52,7 +52,14 @@ class JDBCOptions(
    */
   val asProperties: Properties = {
     val properties = new Properties()
-    parameters.originalMap.foreach { case (k, v) => properties.setProperty(k, v) }
+    parameters.originalMap.foreach { case (k, v) =>
+      // If an option value is `null`, throw a user-friendly error. Keys here cannot be null, as
+      // scala's implementation of Maps prohibits null keys.
+      if (v == null) {
+        throw QueryCompilationErrors.nullArgumentError("JDBC Options", k)
```

Comment:
I don't feel that you use the right error:

```
  "INVALID_PARAMETER_VALUE" : {
    "message" : [
      "The value of parameter(s) <parameter> in <functionName> is invalid:"
    ],
```

Reply:
Maybe you can create a JDBC option error class for this purpose?

Reply:
Yeah, I think we should create a new error class for cases where the data source options value is null. This could be used by other data sources as well.

Reply:
yea a new error class like `NULL_DATA_SOURCE_OPTION` should work.

Reply:
Added a new error class for this purpose.